### ISSUE TITLE
Ambiguity when checking for existence of Backend SGs

### ISSUE THREAD

**Issue by visit1985** [kubernetes-sigs/aws-load-balancer-controller#3953]
When searching for the existence of backend SGs, the controller searches by tags and does not consider the case where multiple SGs with that tags are returned.

To avoid ambiguity, the below function should either search by SG name (which is already present as an input parameter), or return an error when multiple SGs with the given tags are found.

https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/8ba34e27224d1f7a56f0e416086c43b6877c7e65/pkg/networking/backend_sg_provider.go#L284-L310

**Comment by shraddhabang** [comment]
Hey @visit1985 , While configuring the backend sg, the LBC uses a shared backend SG for all the resources which it creates once and applies these tags. So the customer should avoid to use the same tag for other sgs. 
If you want to configure your own backend sg, you can disable this automanagement by setting flag ``--enable-backend-security-group=false``. Then in that case the LBC will configure the SG that you provide. 

More info on SG management : https://kubernetes-sigs.github.io/aws-load-balancer-controller/latest/deploy/security_groups/#backend-security-groups

**Comment by visit1985** [comment]
Yes, right. But even the  it’s better to fail instead if using a random SG via `sgs[0]`.

**Comment by M00nF1sh** [comment]
@visit1985 
We should avoid dependencies on the `name`(which allows us to change naming patterns)
I'm curious why you would run into this issue? since the backendSG is already scoped by the clusterTag & resourceTag and vpcID, there shouldn't be more than one such backendSG exists. Did you created a SG with same set of tags out of the controller?

**Comment by visit1985** [comment]
Yes, someone created it during preparation to migrate to command line flag  `--backend-security-group`.  For some deployments this resulted in a scenario where both SGs are in use.

Adding a raise condition for `len(sgs) > 1` should be backwards compatible, does not create a dependency on `name` and would inform the customer that he should avoid to use the same tag for other SGs.
